### ISSUE TITLE
Failed to compile newmat11 on windows with Visual Studio 2013.

### ISSUE THREAD

**Issue by codetaobeibei** [amy-tabb/RWHEC-Tabb-AhmadYousef#1]
Hi, 
First of all, we are grateful to you for your wonderful work. It is what we need for our robot/world hand/eye calibration. 
However, we use Visual Studio 2013 to make our project, so we are trying to convert your works to our IDE. As a preliminary work, we compiled glog, eigen and ceres successfully. However, we failed to compiled the newmat11 as it is an old library.  So could you please give us some tips on how to build it or use it?

**Comment by amy-tabb** [comment]
Hi there!  It is only by chance I checked this ... apologies for the delay.  

Newmat is maintained by R B Davies, he has some intstructions for building under different systems here:
http://www.robertnz.net/nm11.htm#starting

Relevant details from the page here:
I include make files for various versions of Borland, Microsoft and Intel compilers. With the Borland compiler you will need to edit it to show where you have stored your Borland compiler. For make files for other compilers use my genmake utility. To compile my test and example programs use Borland 5.5 (Builder 5) use

   make -f nm_b55.mak

or with Borland 5.6 (Builder 6) use

   make -f nm_b56.mak

or with Microsoft VC++ 6 or 7 use

   nmake -f nm_m6.mak

There are some more notes in the genmake documentation about using these make files.

Let me know how it goes.  I have a crazy two weeks at work, but it is conceivable that newmat can be removed at Eigen used instead; newmat is just used for matrix manipulations, and I have moved to Eigen recently anyway.

More speedy reply -- amy.tabb@ars.usda.gov

Good luck!
A

**Comment by codetaobeibei** [comment]
Thanks for your kindly reply. In the last few days, we have tried to install all 3rd party library successfully, we used a precompiled newmat11 static library from [here](https://github.com/Vaa3D/v3d_external/tree/master/v3d_main/common_lib/winlib64). Unfortunately,  it still failed to compile the project, because of a bunch of bugs coming-- which may be due to the huge differences between Windows and Linux. So we are trying to do the calibration work on Ubuntu16.04 firstly--following the step by step tutorials you write--without embedding in our project. Later, we may consider the next step about code embedding.
And thanks again! 
Good luck!

**Comment by yulong314** [comment]
I successfully compiled on windows with visualstudio 2013, there is no "newmat11" problems now , only that you need a windows version "dirent.h"

**Comment by amy-tabb** [comment]
@yulong314 Thanks very much for this information! I have updated this whole repository, and removed the newmat11 dependencies and moved over the Eigen from when the original issue was opened.

In any event, the Windows information is useful, since I don't do any development work in Windows, I can update the README with your notes. Thanks again.

And @codetaobeibei -- if you happen to still be working on these problems, the repo was updated over the summer to depend on Eigen for matrix computations instead of newmat.
